tools/syz-declextract/run.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2017 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package main
+
+import (
+	"encoding/json"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/google/syzkaller/pkg/tool"
+)
+
+type compileCommand struct {
+	Arguments []string
+	Directory string
+	File      string
+	Output    string
+}
+
+func run(file string, out chan []byte) {
+	if !strings.HasSuffix(file, ".c") {
+		out <- nil
+		return
+	}
+
+	cmd := exec.Command("syz-declextract", "-p", os.Args[1], file)
+	stdout, err := cmd.Output()
+	if err != nil {
+		tool.Fail(err)
+	}
+	out <- stdout
+}
+
+func main() {
+	if len(os.Args) < 2 {
+		tool.Failf("Please input path to compile_commands.json")
+	}
+	var fileData []byte
+	var err error
+
+	if fileData, err = os.ReadFile(os.Args[1]); err != nil {
+		tool.Fail(err)
+	}
+
+	var c []compileCommand
```

Comment:
While short (and 1 letter in particular) names are common in Go, I would give it a slightly longer name, since it's an important variable and to emphasize plurality: `cmds` or `commands`.

Reply:
I used cmds.